lib/modules/manager/nuget/util.ts

```diff
@@ -52,13 +52,17 @@ export async function getConfiguredRegistries(
   }
 
   const packageSources = nuGetConfig.childNamed('packageSources');
-  if (!packageSources) {
+  const disabledPackageSources = nuGetConfig.childNamed(
+    'disabledPackageSources',
+  );
+
+  if (!packageSources && !disabledPackageSources) {
```

Comment:
```suggestion
  if (!packageSources) {
```

Shouldn't we return here if there are no packageSources? If there's no packageSources, there's nothing to disable. This would allow the if() later down to be reverted

Reply:
>  If there's no packageSources, there's nothing to disable

That's not true, there is always the default registry that could be disabled. That's what we discussed already [here](https://github.com/renovatebot/renovate/pull/32011#discussion_r1814422778).

Reply:
@rarkins Do you agree?

Reply:
I'm not sure I do.

If there are no packageSources, BUT there are disabledPackageSources, what do you expect/recommend the resulting behavior to be?

Reply:
In any case, **I would expect Renovate to respect exactly the content of the _nuget.config_ file in order to match the behavior of NuGet** when NuGet is called outside of Renovate context.

That means:
* If there are no packageSources, BUT there are disabledPackageSources, AND the default source (nuget.org) is NOT among the disabledPackageSources, use nuget.org.
* If there are no packageSources, BUT there are disabledPackageSources, AND the default source (nuget.org) is among the disabledPackageSources, disable nuget.org.
In that case, the user will end up with a configuration with no source, thus NuGet will fail to restore/update the packages. I agree it kind of make no sense to end up with a configuration with no source (or, at least, I don't see a valid use case), but that is what the user will get locally in his repository or on his CI using dotnet / NuGet, so it _does_ make sense to get the same behavior with Renovate.

Reply:
If you leave `registryUrls` blank then Renovate would go back to defaulting to the datasource's default registry (i.e. nuget.org).

So instead you'd need to disable all dependencies by setting a `skipReason`, which can't be overridden. Are you sure you we should try to hard to intentionally fail here, rather than allowing a nuget lookup?

Reply:
Ok, I see what you mean. 👌 

I've pushed a commit with the related changes.